mewarpx/mewarpx/assemblies.py

```diff
@@ -371,8 +387,114 @@ def calculatenormal(self, px, py, pz):
         return nhat
 
 
+class InfCylinderZ(Assembly):
+    """An infinitely long Cylinder pointing in the z-direction with center
+    along the x and y origin line (to support RZ)."""
+    geoms = ['RZ']
+
+    def __init__(self, V, T, WF, name, r_outer, r_inner=0.0):
+        """Basic initialization.
+
+        Arguments:
+            V (float): Voltage (V)
+            T (float): Temperature (K)
+            WF (float): Work function (eV)
+            name (str): Assembly name
+            r_outer (float): The outer radius of the cylinder (m)
+            r_inner (float): The inner radius of the cylinder (m)
+        """
+        super(InfCylinderZ, self).__init__(V=V, T=T, WF=WF, name=name)
+        self.r_inner = r_inner
+        self.r_outer = r_outer
+        self.r_center = (self.r_inner + self.r_outer) / 2.0
+
+        self.center_x = 0.0
+        self.center_y = 0.0
+
+        # sanity check
+        if self.r_outer == 0:
+            raise AttributeError('Cannot have a cylinder with 0 outer radius.')
+
+        # check if the cylinder forms the simulation boundary
+        if self.r_inner == mwxrun.rmax:
+            # set solver boundary potential
+            mwxrun.grid.potential_xmax = self.V
+            # set label to correctly grab scraped particles from the buffer
+            self.scraper_label = 'x_hi'
+        elif self.r_outer == mwxrun.rmin:
+            # set solver boundary potential
+            mwxrun.grid.potential_xmin = self.V
+            # set label to correctly grab scraped particles from the buffer
+            self.scraper_label = 'x_lo'
```

Comment:
I would probably use `np.isclose` with very low `atol` and `rtol`; a script that accidentally computes `rmax` or `rmin` twice with different arithmetic orders could lead to a failure here.